tests/accessor/accessor_implicit_conversions.h

```diff
@@ -0,0 +1,394 @@
+                               Target>(r, get_acc_functor, modify_acc_functor);
+    }
+  }
+
+ public:
+  void operator()(const std::string& type_name) const {
+    const auto section_name =
+        detail::section_name_prototype<AccType, Dimension>::get(type_name)
+            .with("case", "conversion from 'T' to 'const T'")
+            .create();
+    SECTION(section_name) { run_check<DataT, const DataT>(); }
+  }
+};
+
+/**
+ * @brief Validates implicit conversion for host accessor
+ */
+template <typename DataT, typename DimensionT>
+class check_conversion_host {
+  static constexpr auto AccType = accessor_type::host_accessor;
+  static constexpr int Dimension = DimensionT::value;
+
+  static_assert(!std::is_const_v<DataT>, "No need to pass const type here");
+
+  template <typename SrcDataT, sycl::access_mode SrcAccessMode,
+            typename DstDataT, sycl::access_mode DstAccessMode>
+  void run_check() const {
+    constexpr int BufferDimension = (Dimension == 0) ? 1 : Dimension;
+    using src_accessor_t =
+        sycl::host_accessor<SrcDataT, Dimension, SrcAccessMode>;
+    using dst_accessor_t =
+        sycl::host_accessor<DstDataT, Dimension, DstAccessMode>;
+    using src_buffer_t = sycl::buffer<SrcDataT, BufferDimension>;
+
+    constexpr bool has_implicit_conversion_available =
+        std::is_convertible_v<src_accessor_t, dst_accessor_t>;
+
+    // We don't want to fail compilation of entire test every time conversion
+    // is not available for some of accessor template instantiations
+    // So first we check precodition, and only in case of conversion
+    // availability we try to go on with functionality check
+    REQUIRE(has_implicit_conversion_available);
+    // Execution would never pass the 'REQUIRE' call in case there is no
+    // implicit conversion available; still it's better to switch off only
+    // the minimal part of compilation path for better test support
+    using invoke_implicit_conversion_t = std::conditional_t<
+        has_implicit_conversion_available,
+        detail::invoke_implicit_conversion<src_accessor_t, dst_accessor_t>,
+        detail::avoid_implicit_conversion<src_accessor_t>>;
+
+    // We should create source accessor on the host side
+    const auto get_acc_functor = [](src_buffer_t& data_buf) {
+      const src_accessor_t src_acc(data_buf);
+      return src_acc;
+    };
+    // Implicit conversion should be verified within command
+    const invoke_implicit_conversion_t modify_acc_functor;
+
+    if constexpr (Dimension == 0) {
+      check_zero_dim_constructor<AccType, SrcDataT, DstAccessMode>(
+          get_acc_functor, modify_acc_functor);
+    } else {
+      const auto r = util::get_cts_object::range<Dimension>::get(1, 1, 1);
+      check_common_constructor<AccType, SrcDataT, Dimension, DstAccessMode>(
+          r, get_acc_functor, modify_acc_functor);
+    }
+  }
+
+ public:
+  void operator()(const std::string& type_name) const {
```

Comment:
The code looks very close to the `generic` pointer case. Can we refactor it to avoid duplication?

Reply:
generic code has been moved into a separate function in this commit: 11a405dc31cd1386628d25aed171ed52abad3b9a